images/win/scripts/ImageHelpers/VisualStudioHelpers.ps1

```diff
@@ -61,20 +61,57 @@ Function Install-VisualStudio
 }
 
 function Get-VsCatalogJsonPath {
-    $instanceFolder = Get-Item "C:\ProgramData\Microsoft\VisualStudio\Packages\_Instances\*" | Select-Object -First 1
-    return Join-Path $instanceFolder.FullName "catalog.json"
+    $instanceFolder = "C:\ProgramData\Microsoft\VisualStudio\Packages\_Instances\" + (Get-VisualStudioProduct -ProductType "VisualStudio").InstanceId
+    return Join-Path $instanceFolder "catalog.json"
 }
 
-function Get-VisualStudioPath {
-    return (Get-VSSetupInstance | Select-VSSetupInstance -Product *).InstallationPath
-}
+function Get-VisualStudioProduct {
+    Param
+    (
+        [Parameter(Mandatory)]
+        [ValidateSet('VisualStudio','BuildTools')]
+        [String] $ProductType
+    )
 
-function Get-VisualStudioPackages {
-    return (Get-VSSetupInstance | Select-VSSetupInstance -Product *).Packages
+    if ($ProductType -eq "VisualStudio")
+    {
+        $VSSelectionType = "*Enterprise*"
+    }
+    elseif ($ProductType -eq "BuildTools")
+    {
+        $VSSelectionType = "*Build*"
+    }
+    return Get-VSSetupInstance | Select-VSSetupInstance -Product * | Where-Object -Property DisplayName -like $VSSelectionType
 }
 
 function Get-VisualStudioComponents {
-    Get-VisualStudioPackages | Where-Object type -in 'Component', 'Workload' |
+    Param
+    (
+        [Parameter(Mandatory)]
+        [String] $ProductType
+    )
+    (Get-VisualStudioProduct -ProductType $ProductType).Packages | Where-Object type -in 'Component', 'Workload' |
     Sort-Object Id, Version | Select-Object @{n = 'Package'; e = {$_.Id}}, Version |
     Where-Object { $_.Package -notmatch "[0-9a-fA-F]{8}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{12}" }
-}
\ No newline at end of file
+}
+
+function Get-VSExtensionVersion
+{
+    Param
+    (
+        [Parameter(Mandatory=$true)]
+        [string] $packageName
+    )
+
+    $instanceFolders = "C:\ProgramData\Microsoft\VisualStudio\Packages\_Instances\" + (Get-VisualStudioProduct -ProductType "VisualStudio").InstanceId
```

Comment:
Why don't we create a separate function to get instanceFolders due to remove duplicate code in 106 and 64 string?

Reply:
I'll add corresponding function, it seems good idea